packages/server/graphql/public/types/AddedNotification.ts

```diff
@@ -0,0 +1,20 @@
+import {getUserId} from '../../../utils/authorization'
+import {AddedNotificationResolvers, ResolversTypes} from '../resolverTypes'
+
+export type AddedNotificationSource = {addedNotificationId: string}
+
+const AddedNotification: AddedNotificationResolvers = {
+  addedNotification: async ({addedNotificationId}, _args: unknown, {dataLoader, authToken}) => {
+    const viewerId = getUserId(authToken)
+    const notification = await dataLoader.get('notifications').load(addedNotificationId)
+    if (notification.userId !== viewerId) {
+      throw new Error(
+        `Viewer ID does not match notification user ID: notification ${addedNotificationId} for user ${notification.userId} published to user ${viewerId}`
+      )
+    }
+
+    return notification as ResolversTypes['Notification']
```

Comment:
Ah good call! I took a slightly different approach by adding each individual notification type to the codegen, so new notifications won't need to be added to an `AnyNotification` list somewhere and the union type is enforced automatically by SDL instead of manually by us.

Reply:
that's cleaner :+1: